odoo/models.py

```diff
@@ -6830,9 +6830,26 @@ def diff(self, other, force=False):
                 for name, subnames in self['<tree>'].items():
                     if name == 'id':
                         continue
+                    field = record._fields[name]
```

Comment:
Seems like a very complicated fix and honestly I don't understand everything, but I can at least ask if this should be tested with a unit test? :smile: